packages/core/src/indexing-store/utils/encoding.ts

```diff
@@ -63,140 +188,179 @@ export function encodeRow(
       );
     }
 
-    instance[columnName] = encodeValue(value, column, encoding);
+    validateField({ value, column, schema, skipValidate });
   }
-
-  return instance;
 }
 
-/**
- * Convert a user-land value into a database-ready value.
- */
-export function encodeValue(
-  value: UserColumn,
-  column: Column,
-  encoding: "sqlite" | "postgres",
-): DatabaseColumn {
-  if (isEnumColumn(column)) {
-    if (isOptionalColumn(column) && (value === undefined || value === null)) {
-      return null;
-    }
+function validateField({
+  value,
+  column,
+  schema,
+  skipValidate,
+}: {
+  value: UserColumn;
+  column: Column;
+  schema: Schema;
+  skipValidate: boolean;
```

Comment:
After reading through this function - should this option be called `skipConstraintValidation` or `skipDatabaseValidation`?

It seems like this function _always_ validates encoding issues - it's not clear to me which category of validations get skipped when `skipValidate` is true.

Reply:
`skipValidate` is meant to mean "skip any extra validation that would be caught by the database". Because of the very small overhead of validating in js runtime compared to a db query, I decided to never skip validation and remove the parameter.